listenbrainz/mbid_mapping/mapping/mb_metadata_cache.py

```diff
@@ -396,51 +399,223 @@ def config_postgres_join_limit(self, curs):
+                    ON rel.release_group = rg.id
+             LEFT JOIN cover_art_archive.cover_art caa
+                    ON caa.release = rel.id
+             LEFT JOIN cover_art_archive.cover_art_type cat
+                    ON cat.id = caa.id
+                 WHERE  rel.last_updated > %(timestamp)s
+                    OR   rg.last_updated > %(timestamp)s
+                    OR (caa.date_uploaded > %(timestamp)s AND (type_id = 1 OR type_id IS NULL))
+            ) SELECT r.gid
+                FROM recording r
+                JOIN track t
+                  ON t.recording = r.id
+                JOIN medium m
+                  ON m.id = t.medium
+                JOIN release_mbids rm
+                  ON rm.id = m.release
+        """
 
-        conn = self.lb_conn if self.lb_conn is not None else self.mb_conn
         try:
-            with conn.cursor() as curs:
-                query = f"""UPDATE {self.table_name}
-                               SET dirty = 't'
-                             WHERE recording_mbid IN %s
-                                OR %s && artist_mbids
-                                OR release_mbid IN %s
-                        """
-                curs.execute(query, (tuple(recording_mbids), artist_mbids, tuple(release_mbids)))
-                conn.commit()
+            with self.mb_conn.cursor() as curs:
+                recording_mbids = set()
+
+                log("mb metadata cache: querying recording mbids to update")
+                curs.execute(recording_mbids_query, {"timestamp": timestamp})
+                for row in curs.fetchall():
+                    recording_mbids.add(row[0])
+
+                log("mb metadata cache: querying artist mbids to update")
+                curs.execute(artist_mbids_query, {"timestamp": timestamp})
+                for row in curs.fetchall():
+                    recording_mbids.add(row[0])
+
+                log("mb metadata cache: querying release mbids to update")
+                curs.execute(release_mbids_query, {"timestamp": timestamp})
+                for row in curs.fetchall():
+                    recording_mbids.add(row[0])
 
+                return recording_mbids
         except psycopg2.errors.OperationalError as err:
-            log("mb metadata cache: cannot mark rows as dirty", err)
-            conn.rollback()
-            raise
+            log("mb metadata cache: cannot query rows for update", err)
+            return None
 
-    def update_dirty_cache_items(self):
+    def update_dirty_cache_items(self, recording_mbids: Set[uuid.UUID]):
         """Refresh any dirty items in the mb_metadata_cache table.
 
         This process first looks for all recording MIBDs which are dirty, gets updated metadata for them, and then
         in batches deletes the dirty rows and inserts the updated ones.
         """
-        dirty_query = f"""
-            SELECT recording_mbid
-              FROM {self.table_name}
-             WHERE dirty = 't'
```

Comment:
does this mean that the dirty column is no longer used? Does it make sense to do this process in 2 steps - get mbids and mark as dirty, and then update?

Reply:
Yes, dirty column is no longer used. We already have the list of recording mbids at hand so figured no need to mark dirty then query and update. I guess we could do that if we wanted to manually marks rows as dirty and want the update process to consider those as well?